testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6268,6 +6268,18 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
     f.checkScalar("rand_integer(2, 11)", 1, "INTEGER NOT NULL");
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-6283">
+   * [CALCITE-6283] Function array_append with a NULL array argument crashes with
+   * NullPointerException</a>. */
+  @Test void testArrayNullFunc() {
+    final SqlOperatorFixture f = fixture().withLibrary(SqlLibrary.SPARK);
+    f.checkNull("array_append(null, 2)");
+    f.checkNull("array_prepend(null, 2)");
+    f.checkNull("array_remove(null, 2)");
+    f.checkNull("array_contains(null, 2)");
```

Comment:
Do we have tests for appending null elements

Reply:
@tanclary Yes, similar to:
 f.checkNull("array_append(cast(null as integer array), 1)");

Reply:
I meant a test where the element is null, not the array. But i found that we do so that's good.

Reply:
I tested it in spark3.4. It seems that the first parameter of array_append is null. Why do we need to change it like this?

Reply:
![捕获](https://github.com/apache/calcite/assets/77189278/a081d503-71a1-4222-b3f1-70a35b0e4a22)

Reply:
I don't understand your question

Reply:
My problem is that array_append is to adapt to spark, but spark’s array_append(null, 2) will report an error.

Reply:
What I mean is that spark3.4 version does not support array_append(null, 1), it will report an error. Currently, array_append and array_prepend are adapted for spark in calcite. If spark does not support array_append(null , 1), why do we need to return null? I am not sure whether higher versions of spark support it.
I don't know if my opinion is useful, if it is problematic, please ignore it. But I still want to express my opinion. This is open source, thank you.
![捕获](https://github.com/apache/calcite/assets/77189278/dfefad4e-45d0-431a-a55c-df340878ba16)

Reply:
Ah. Yes I agree with you. What do you think @mihaibudiu ?

Reply:
I have changed the type checker to reject NULL literals for arrays.
However, it seems to be that Spark is inconsistent, since it allows NULL values for arrays at runtime, but not at compile time. This is probably because Spark uses a different type checking/inference algorithm.

Reply:
> I have changed the type checker to reject NULL literals for arrays. However, it seems to be that Spark is inconsistent, since it allows NULL values for arrays at runtime, but not at compile time. This is probably because Spark uses a different type checking/inference algorithm.

spark sql, after generating the Unresolved Logical Plan, runs a series of rules to turn it into the Logical Plan, which solves some type conversion issues there, and the type handling of array_append(null,2) seems to be in that part